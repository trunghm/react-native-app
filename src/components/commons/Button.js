import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Colors from '../themes/Colors';
import Fonts from '../themes/Fonts';

import {
  StyleSheet,
  Text,
  View,
  TouchableHighlight
} from 'react-native';

export default class Button extends Component {

  constructor(props) {
    super(props);
  }

  render() {
    let props = {...this.props};
    const {isShowRadiusButton} = this.props;
    const radiusStyle = isShowRadiusButton ? styles.radiusButton : {};
    return (
      <View style={styles.container}>
        <TouchableHighlight
          onPress={this.props.onPress}
          underlayColor={Colors.pressed_color}
                            style={[styles.buttonWrapper, props.disabled?styles.disabled:null, radiusStyle]}
        >
          <Text style={styles.text}>{this.props.text.toUpperCase()}</Text>
        </TouchableHighlight>
      </View>);
  }

  static propTypes = {
    text: PropTypes.string.isRequired,
    isShowRadiusButton: PropTypes.bool,
  };
  // , radiusStyle

  static defaultProps = {
    text: 'Button',
    disabled: false,
    onPress: () => {},
    isShowRadiusButton: false,
  };
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  text: {
    fontSize: 14,
    color: '#fff',
    fontWeight: 'bold',
    letterSpacing: -0.34,
  },
  disabled: {
    backgroundColor: Colors.button_disbaled_color,
  },
  buttonWrapper: {
    backgroundColor: Colors.primary_color,
    height: 50,
    alignItems: 'center',
    justifyContent: 'center'
  },
  radiusButton: {
    borderRadius: 5,
    marginBottom: 24,
    marginRight: 16,
    marginLeft: 16
  }
});
