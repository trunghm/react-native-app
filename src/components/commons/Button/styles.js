import EStyleSheet from "react-native-extended-stylesheet";
import GlobalStore from "../../../themes/GlobalStore";

const styles = () => EStyleSheet.create({

  wrap:{
    paddingLeft: 15,
    paddingRight: 15,
    alignSelf:"flex-start",
  },

  general:{
    maxHeight:50, minHeight:50,
    flex: 1,
    justifyContent:"center",
    alignItems:"center",
    backgroundColor: GlobalStore.color.primary,
  },

  text: {
    fontSize: 14,
    color: GlobalStore.color.buttonTitle,
    fontWeight: 'bold',
  },
  disabled: {
    backgroundColor: GlobalStore.color.buttonBgDisable,
  },

  round: {
    borderRadius: 45
  }
});

export default styles;
