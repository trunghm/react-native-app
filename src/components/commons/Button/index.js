import React, {Component} from 'react';
import styles from "./styles";
import {
  Text,
  View,
  TouchableOpacity
} from 'react-native';

export default function Button({onPress = ()=>{}, text ="", round, disabled, full} = {}) {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[full ? styles().fullType : styles().wrap, styles().general, round && styles().round]}
    >
      <Text style={styles().text}>{text.toUpperCase()}</Text>
    </TouchableOpacity>
  )
}





