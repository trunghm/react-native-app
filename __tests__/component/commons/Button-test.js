/**
 * @format
 */

import 'react-native';
import React from 'react';
import Button from '../../../src/components/commons/Button';

// Note: test renderer must be required after react-native.
import { shallow } from 'enzyme';
import EStyleSheet from 'react-native-extended-stylesheet';

beforeAll(() => {
  EStyleSheet.build({

  });
});
describe('Testing  Button component', () => {

  it('renders as expected', () => {
    const wrapper = shallow(
      <Button text ="Ex: Button" />
    );

    expect(wrapper).toMatchSnapshot();

  });

});
