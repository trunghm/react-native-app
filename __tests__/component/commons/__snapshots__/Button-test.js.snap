// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  Button component renders as expected 1`] = `ShallowWrapper {}`;

exports[`Testing  Button component renders snapshot as expected 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#1496bb",
      "flex": 1,
      "justifyContent": "center",
      "maxHeight": 50,
      "minHeight": 50,
      "opacity": 1,
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    EX: BUTTON
  </Text>
</View>
`;
